pkg/cmd/pipeline/start_test.go

```diff
@@ -105,7 +150,254 @@ func Test_start_trigger(t *testing.T) {
 	}
 }
 
-func Test_start_trigger_client_error(t *testing.T) {
+func Test_start_pipeline_last(t *testing.T) {
+
+	pipelineName := "test-pipeline"
+
+	ps := []*v1alpha1.Pipeline{
+		tb.Pipeline(pipelineName, "ns",
+			tb.PipelineSpec(
+				tb.PipelineDeclaredResource("git-repo", "git"),
+				tb.PipelineDeclaredResource("build-image", "image"),
+				tb.PipelineParam("pipeline-param-1", tb.PipelineParamDefault("somethingdifferent-1")),
+				tb.PipelineParam("rev-param", tb.PipelineParamDefault("revision")),
+				tb.PipelineTask("unit-test-1", "unit-test-task",
+					tb.PipelineTaskInputResource("workspace", "git-repo"),
+					tb.PipelineTaskOutputResource("image-to-use", "best-image"),
+					tb.PipelineTaskOutputResource("workspace", "git-repo"),
+				),
+			), // spec
+		), // pipeline
+	}
+
+	prs := []*v1alpha1.PipelineRun{
+		tb.PipelineRun("test-pipeline-run-123", "ns",
+			tb.PipelineRunLabel("tekton.dev/pipeline", pipelineName),
+			tb.PipelineRunSpec(pipelineName,
+				tb.PipelineRunServiceAccount("test-sa"),
+				tb.PipelineRunResourceBinding("git-repo", tb.PipelineResourceBindingRef("some-repo")),
+				tb.PipelineRunResourceBinding("build-image", tb.PipelineResourceBindingRef("some-image")),
+				tb.PipelineRunParam("pipeline-param-1", "somethingmorefun"),
+				tb.PipelineRunParam("rev-param", "revision1"),
+			),
+		),
+	}
+
+	objs := []runtime.Object{ps[0], prs[0]}
+	pClient, tracker := NewPipelineClient(objs...)
+
+	cs := test.Clients{
+		Pipeline: pClient,
+		Kube:     fakekubeclientset.NewSimpleClientset(),
+	}
+
+	p := &tu.Params{Tekton: cs.Pipeline, Kube: cs.Kube}
+
+	cs.Pipeline.PrependReactor("create", "pipelineruns", func(action k8stest.Action) (bool, runtime.Object, error) {
```

Comment:
Shouldn't this be in `NewPipelineClient` instead ?

Reply:
Nope. It can be different for different test cases.

Reply:
Is it right now ? My point is, if it's 80% time the same, then let's have this in `NewPipelineClient` and have a way to override the default on demand ; something like : 

```go
type prependReactorFn func(*fakepipelineclientset.Clientset, verb, resource string, reaction ReactionFunc)

func newPipelineClient(prf prependReactorFn objs ...runtime.Object) (*fakepipelineclientset.Clientset, k8stest.ObjectTracker) { 
	// […]
	c := &fakepipelineclientset.Clientset{}
	c.AddReactor("*", "*", k8stest.ObjectReaction(o))
	if prf != nil {
		pfr(c)
	} else {
		// default prepend reactor
		c.PrependReactor(…)
	}
}
// By default
pClient, tracker := newPipelineClient(nil, objs...)

// With a custom one
func listPipelineRunsError(c *fakepipelineclientset.Clientset, verb, resource string, reaction ReactionFunc) {
	c.PrependReactor("list", "pipelineruns", func(action k8stest.Action) (bool, runtime.Object, error) {
		return true, nil, errors.New("test generated error")
	}
}
pClient, tracker := newPipeline(listPipelineRunsError, objs...)
```

Or even better, using a builder pattern and sane defaults, when creating the client.